src/Stream/BiDirectionalSeekableReader.h

```diff
@@ -1,23 +1,14 @@
 #pragma once
 
-#include "Reader.h"
+#include "ForwardSeekableReader.h"
 #include <cstdint>
 
 namespace Stream
 {
-	class BiDirectionalSeekableReader : public Reader {
+	class BiDirectionalSeekableReader : public ForwardSeekableReader {
 	public:
-		// Get the size of the stream
-		virtual uint64_t Length() = 0;
-
-		// Get the current position of the stream
-		virtual uint64_t Position() = 0;
-
 		// Seek to absolute position, given as offset from beginning of stream
 		virtual void Seek(uint64_t position) = 0;
```

Comment:
We should probably move absolute seek last. Having random access is more specific than only forward or backward seeking. We could even create a `RandomAccessReader` class if we wanted.